server/src/routes/assignment.ts

```diff
@@ -0,0 +1,126 @@
+// SPDX-FileCopyrightText: 2023 The Aalto Grades Developers
+//
+// SPDX-License-Identifier: MIT
+
+import express, { Router } from 'express';
+
+import { addAssignment, updateAssignment } from '../controllers/courseAssignment';
+import { handleInvalidRequestJson } from '../middleware';
+import { controllerDispatcher } from '../middleware/errorHandler';
+
+export const router: Router = Router();
+
+/**
+ * @swagger
+ * definitions:
+ *   AddAssignment:
+ *     type: object
+ *     description: Assignment Information for adding a new assignment.
+ *     properties:
+ *       courseInstanceId:
+ *         type: number
+ *         description: Course instance ID to which the assignment belongs to.
+ *       name:
+ *         type: string
+ *         description: Assignment name.
+ *       executionDate:
+ *         type: string
+ *         description: Date when the assignment is completed (e.g. deadline date or exam date).
+ *       expiryDate:
+ *         type: string
+ *         description: Date when the assignment expires.
+ *   Assignment:
+ *     description: Current assignment Information.
+ *     allOf:
+ *       - type: object
+ *         properties:
+ *           id:
+ *             type: number
+ *             required: true
+ *             description: Assignment ID.
+ *       - $ref: '#/definitions/AddAssignment'
+ */
+
+/**
+ * @swagger
+ * /v1/assignment:
+ *   post:
+ *     tags: [Assignment]
+ *     description: Add a new assignment.
+ *     requestBody:
+ *       description: New assignment data.
+ *       content:
+ *         application/json:
+ *           schema:
+ *             $ref: '#/definitions/AddAssignment'
+ *     responses:
+ *       200:
+ *         description: New assignment succesfully added.
+ *         content:
+ *           application/json:
+ *             schema:
+ *               $ref: '#/definitions/Assignment'
+ *       400:
+ *         description: Creation failed, due to malformed/missing parameters.
+ *         content:
+ *           application/json:
+ *             schema:
+ *               $ref: '#/definitions/Failure'
+ *       404:
+ *         description: Course instance does not exist.
+ *         content:
+ *           application/json:
+ *             schema:
+ *               $ref: '#/definitions/Failure'
+ */
+router.post(
+  '/v1/assignment',
```

Comment:
Since course assignments are instance specific, this route could maybe be `/v1/courses/instances/:instanceId/assignments` and the instance ID can be removed from the request body to be consistent with other routes. Assignments could also be a candidate for using composite keys in the next sprint.

I guess this is fine for now though since we don't really have time to collectively decide on specifications for these routes at the moment.

Reply:
Thinking about this now I don't really like the half measure of putting `/v1/courses/instances` at the start of this route without the course ID, since we would also have routes which do have it like `/v1/courses/:courseId/instances`, which seems messy to me.

My worry with using composite keys for assignments is that we would end up with routes like `/v1/courses/:courseId/instances/:instanceId/assignments/:assignmentId` which might be too complex for its own good.

But on the other hand if we want to get all assignments of a course instance, without composite keys we would still somehow need to include the course instance ID in the URL, like `/v1/course-instances/:instanceId/assignments`, but then we run into the problem of inconsistency like with the `/v1/courses/:courseId/instances` and `/v1/instances` problem we discussed before.

You can ignore this comment for this PR, but we should consider this later.